src/tinflate.c

```diff
@@ -398,6 +398,10 @@ static int tinf_inflate_block_data(TINF_DATA *d, TINF_TREE *lt, TINF_TREE *dt)
         }
     } else {
         if(d->dest >= d->edest) return TINF_DATA_ERROR;
+        if(d->lzOff >= 0) return TINF_DATA_ERROR;
```

Comment:
Why these checks are being put here? What about the "true" block of the "if" statement?

Is there an example of "specially crafted input data" (re: commit message)?

Reply:
In the zip of crashing inputs, one example that is fixed by this commit is:
explore-findings/crashes/id:000022,sig:11,src:000000,op:havoc,rep:4

I think the specific change is nonsesnse for the 'true' arm of the loop, because in that side, `lzOff` is used to index into `dict_ring` rather than relative to the output data pointer `dest`.  I do not have any testcases that show a problem with invalid access into `dict_ring`.

Reply:
So, the problem here is someone could read uninitialized data from dict_ring with such a specially crafted offset. But yep, we'll consider that to be a client problem (should not supply uninitialized dict buffer).

Reply:
> I think the specific change is nonsesnse for the 'true' arm of the loop, because in that side, lzOff is used to index into dict_ring rather than relative to the output data pointer dest. I do not have any testcases that show a problem with invalid access into dict_ring.

Please mind how this was addressed in https://github.com/pfalcon/uzlib/commit/b4eb23c7c35e3cf830585c366e9b461e7dcb9e52 . While UBSan doesn't complain about it, we still must think what happens in the other branch (if we the top-level goal is improving security, not just making a particular tool not complain), and "let know" a user what to keep in mind there.